src/EnvironmentManagement/EnvironmentManager.py

```diff
@@ -342,13 +355,31 @@ def manage_removal_from_game_for(self,
         """
         player_was_removed = (self.__remove_player_from_waitlist(player_id) or
                               self.__remove_player_from_vehicle(player_id))
-
         if player_was_removed:
             self._publish_removed_player(player_id=player_id, reason=reason)
             self.update_staff_ui()
+            self.manage_bot_safe_mode()
             return True
         else:
             return False
+
+    def manage_bot_safe_mode(self)-> None:
+        is_no_player_left = True
+        for v_id in self._active_physical_cars:
+            v = self.get_vehicle_by_vehicle_id(v_id)
+            if v.get_player_id() != None:
+                is_no_player_left = False
+        for v_id in self._active_virtual_cars:
+            v = self.get_vehicle_by_vehicle_id(v_id)
+            print(v.get_player_id)
+            if v.get_player_id() != None:
+                is_no_player_left = False
```

Comment:
As soon as a player has been found, you could break out of the loops checking for players.